PhysicsTools/PatAlgos/src/MuonMvaIDEstimator.cc

```diff
@@ -0,0 +1,88 @@
+#include "PhysicsTools/PatAlgos/interface/MuonMvaIDEstimator.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+using namespace pat;
+using namespace cms::Ort;
+
+MuonMvaIDEstimator::MuonMvaIDEstimator(const edm::FileInPath &weightsfile) {
+  randomForest_ = std::make_unique<ONNXRuntime>(weightsfile.fullPath());
+  std::cout << randomForest_.get() << std::endl;
+}
+
+void MuonMvaIDEstimator::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::FileInPath>("mvaIDTrainingFile", edm::FileInPath("RecoMuon/MuonIdentification/data/mvaID.onnx"));
+  desc.add<std::vector<std::string>>("flav_names",
+                                     std::vector<std::string>{
+                                         "probBAD",
+                                         "probGOOD",
+                                     });
+
+  descriptions.addWithDefaultLabel(desc);
+}
+
+std::unique_ptr<cms::Ort::ONNXRuntime> MuonMvaIDEstimator::initializeGlobalCache(const edm::ParameterSet &iConfig) {
```

Comment:
I'm confused where this `MuonMvaIDEstimator::initializeGlobalCache` function is called, and how does it relate with what you do in the constructor `MuonMvaIDEstimator::MuonMvaIDEstimator`. `MuonMvaIDEstimator` is not an `edm::stream::EDProducer`, so it has no shared state across threads.

The actual initialization of the Muon MVA happens in `PATMuonProducer::initializeGlobalCache`.

Reply:
I have removed the function.